lms/djangoapps/courseware/module_render.py

```diff
@@ -121,6 +121,15 @@ def _get_required_content(course, user):
             if milestone_path.get('content') and len(milestone_path['content']):
                 for content in milestone_path['content']:
                     required_content.append(content)
+
+    can_skip_entrance_exam = user_can_skip_entrance_exam(user, course.id)
+    # check if required_content has any entrance exam and user is allowed to skip it
+    # then remove it from required content
+    if required_content and getattr(course, 'entrance_exam_enabled', False) and can_skip_entrance_exam:
+        descriptors = [modulestore().get_item(UsageKey.from_string(content)) for content in required_content]
+        entrance_exam_contents = [unicode(descriptor.location)
+                                  for descriptor in descriptors if descriptor.is_entrance_exam]
+        required_content = list(set(required_content) - set(entrance_exam_contents))
```

Comment:
Yes, we want to keep the door open for other types of content down the road which may gate access to some/all of the course content.  For example, a milestone requirement could be added in the middle of a student's procession through a course, which would then block access to the remaining content until they have completed the required content and collected the resulting milestone.